src/DynamoCoreWpf/ViewModels/Search/NodeAutoCompleteSearchViewModel.cs

```diff
@@ -48,6 +48,23 @@ internal void PopulateAutoCompleteCandidates()
             });
         }
 
+        internal void SearchAutoCompleteCandidates(string input)
+        {
+            if (PortViewModel == null) return;
+
+            var searchElements = GetMatchingNodes();
```

Comment:
I think this has the potential to be very slow - this seems to be recalculating the matched nodes every time a user enters a new character...
I think this should be calculated only once

Reply:
Yes, I have changed this. Will be updating with new commits. Running into some errors with some workflows.

Reply:
Yes, I agree, the matched nodes should have already been calculated just before the autocomplete popup shows up so this looks wasteful to re-calculate the matched nodes each time you type a search character. Can we cache them and re-use that cache for search?

Reply:
Also, I think this API has been renamed to `GetMatchingSearchElements` in latest master.